### ISSUE TITLE
Asynchttpserver respond function error : hasKey called on a nil httpHeader

### ISSUE THREAD

**Issue by mtha790** [nim-lang/Nim#14206]
Function respond try to call hasKey method on a nil httpHeader. 

### Example
```nim
import asynchttpserver, asyncdispatch

var server = newAsyncHttpServer()
proc cb(req: Request) {.async.} =
  await req.respond(Http200, "Hello World")

waitFor server.serve(Port(8080), cb)
```

### Current Output
```
..
/usr/local/Cellar/nim/1.2.0/nim/lib/pure/asynchttpserver.nim(103) respond
/usr/local/Cellar/nim/1.2.0/nim/lib/pure/httpcore.nim(180) hasKey
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
```

### Possible Solution

* In file asynchttpserver.nim line 103, hasKey is called on a nil httpHeader and generate a SIGSEGV error. 
* A possible fix to this problem is to include the block calling "hasKey" in the condition "headers not nil" : (asynchttpserver.nim line 100 - 106)
```nim
if headers != nil:
    msg.addHeaders(headers)    
    if not headers.hasKey("Content-Length"):
      msg.add("Content-Length: ")
      # this particular way saves allocations:
      msg.addInt content.len
```

### Additional Information

* osx catalina 10.15.4 

```
$ nim -v
Nim Compiler Version 1.2.0 [MacOSX: amd64]
```

**Comment by Yardanico** [comment]
That's already fixed in devel and backported to 1.2 branch (and it'll be in the next 1.2.x release or 1.4), can be closed

**Comment by ghost** [comment]
Duplicate of https://github.com/nim-lang/Nim/issues/13866.
It was fixed in https://github.com/nim-lang/Nim/pull/13867 and will be in the next 1.2.x release (or 1.4 if 1.2.x doesn't happen)
